cgrok/grok.go

```diff
@@ -57,12 +65,16 @@ func New() *Grok {
    With other strings there's a risk of getting the wrong result if memory has been freed and reused. */
 func (grok *Grok) gostringn(str *C.char, len C.int) string {
 	ptr := uintptr(unsafe.Pointer(str))
+	grok.stringCacheLock.RLock()
```

Comment:
Can you make a PR for this improvement? It's a good idea but we're hitting diminishing returns and I want to limit how much time I spend on Grok performance